app/vep/models/upload_vcf_files.py

```diff
@@ -0,0 +1,87 @@
+"""
+See the NOTICE file distributed with this work for additional information
+regarding copyright ownership.
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+"""
+import os
+import logging
+import tempfile
+import shutil
+
+from starlette.requests import ClientDisconnect
+
+from streaming_form_data import StreamingFormDataParser
+from streaming_form_data.targets import FileTarget, ValueTarget
+import streaming_form_data
+
+from core.config import VCF_UPLOAD_DIRECTORY
+
+MAX_FILE_SIZE = 1024 * 1024 * 1024 * 2 # 2GB
+MAX_REQUEST_BODY_SIZE = MAX_FILE_SIZE + 1024
+
+class MaxBodySizeException(Exception):
+    def __init__(self, body_len: str):
+        self.body_len = body_len
+
+
+class MaxBodySizeValidator:
+    def __init__(self, max_size: int):
+        self.body_len = 0
+        self.max_size = max_size
+
+    def __call__(self, chunk: bytes):
+        self.body_len += len(chunk)
+        if self.body_len > self.max_size:
+            raise MaxBodySizeException(body_len=self.body_len)
+
+
+class Streamer:
+    def __init__(self, request):
+        self.request = request
+        self.filename = self.request.headers.get('Filename', None)
+        self.file_name_validator(self.filename)
+        self.temp_dir = tempfile.mkdtemp(dir=VCF_UPLOAD_DIRECTORY)
+        self.filepath = os.path.join(str(self.temp_dir), os.path.basename(self.filename))
+        self._input_file = FileTarget(self.filepath)
```

Comment:
I think leading underscore is not needed here as `_input_file` is not a private variable - it will be sent to Nextflow pipeline (via ini conf file).

Reply:
For the streaming operation this specific variable doesn't need to be accessible outside of the class since it's only used to stream the data to the file. We have filepath available for that.

Reply:
Ah yes.